src/stdatamodels/fits_support.py

```diff
@@ -476,21 +476,27 @@ def callback(schema, path, combiner, ctx, recurse):
                 hdulist, fits_keyword, schema,
                 ctx.get('hdu_index'), known_keywords)
 
-            if result is None:
+            if result is None and context._validate_on_assignment:
                 validate.value_change(path, result, schema, context)
             else:
-                if validate.value_change(path, result, schema, context):
+                if context._validate_on_assignment:
+                    if validate.value_change(path, result, schema, context):
+                        properties.put_value(path, result, tree)
+                else:
                     properties.put_value(path, result, tree)
 
         elif 'fits_hdu' in schema and (
                 'max_ndim' in schema or 'ndim' in schema or 'datatype' in schema):
             result = _fits_array_loader(
                 hdulist, schema, ctx.get('hdu_index'), known_datas)
 
-            if result is None:
+            if result is None and context._validate_on_assignment:
```

Comment:
This if/else block is identical to the one in the previous case, is it work defining a little inner function to do this?